### PR TITLE
i47–add functionality to build a projects object

### PR DESCRIPTION
**Pull request by eshwaribhide**

### CODE REVIEW

File: server-code/Projects.js [hunk 1]
```diff
@@ -11,3 +11,36 @@ function getAllProjects() {
...
+    }
+    if (rowData.length == 0) {
+        throw "No Project data found."
+    }
+    var project = {
+        wbsRowIdx: wbsRow,
+        wbsNum: rowData[wbsColIdx],
+        name: rowData[findIdx("Name", headers)],
+        lead: rowData[findIdx("Project Lead", headers)],
+        name: rowData[findIdx("Project Manager", headers)],
```
Comment (jamescd18) [inline server-code/Projects.js:43]: doesn't this override the previously defined "name" attribute for the project?

```suggestion
        projectManager: rowData[findIdx("Project Manager", headers)],
```

File: server-code/Projects.js [hunk 2]
```diff
@@ -11,3 +11,36 @@ function getAllProjects() {
...
+        }
+    }
+    if (rowData.length == 0) {
+        throw "No Project data found."
+    }
+    var project = {
+        wbsRowIdx: wbsRow,
+        wbsNum: rowData[wbsColIdx],
+        name: rowData[findIdx("Name", headers)],
+        lead: rowData[findIdx("Project Lead", headers)],
```
Comment (jamescd18) [inline server-code/Projects.js:42]:
```suggestion
        projectLead: rowData[findIdx("Project Lead", headers)],
```
just for clarity and to match below

**Review by jamescd18 — CHANGES_REQUESTED** [review]
this is looking pretty great! just some minor edits to the names in the project data object. also, I'd love to have #70 go through before this one, since #70 will add additional fields to the project data object.

File: server-code/Projects.js (hunk 1 above)
Comment (eshwaribhide) [reply server-code/Projects.js:43]: lol dumb mistake

**Review by kevinyu328 — APPROVED** [review]
Looks good to me too! I'll wait till #70 is finished to add the additional fields and merge the pull request.

File: server-code/Projects.js [hunk 3]
```diff
@@ -11,3 +11,38 @@ function getAllProjects() {
...
+    if (rowData.length == 0) {
+        throw "No Project data found."
+    }
+    var project = {
+        wbsRowIdx: wbsRow,
+        wbsNum: rowData[wbsColIdx],
+        name: rowData[findIdx("Name", headers)],
+        projectLead: rowData[findIdx("Project Lead", headers)],
+        projectManager: rowData[findIdx("Project Manager", headers)],
+        slideDeck: rowData[findIdx("Slide Deck", headers)],
```
Comment (jamescd18) [inline server-code/Projects.js:44]:
```suggestion
        slideDeckLink: rowData[findIdx("Slide Deck", headers)],
```

File: server-code/Projects.js [hunk 4]
```diff
@@ -11,3 +11,38 @@ function getAllProjects() {
...
+        bom: rowData[findIdx("BOM", headers)],
```
Comment (jamescd18) [inline server-code/Projects.js:45]:
```suggestion
        bomLink: rowData[findIdx("BOM", headers)],
```

**Review by jamescd18 — CHANGES_REQUESTED** [review]
small things for specificity

**Review by jamescd18 — COMMENTED** [review]
Looks good to me! pending Kevin's review

**Review by kevinyu328 — APPROVED** [review]
Nice job!

**Pull request merged**